src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Snapshot/DependenciesSnapshot.cs

```diff
@@ -165,13 +165,10 @@ void SyncTargetFrameworks()
             Requires.NotNull(activeTargetFramework, nameof(activeTargetFramework));
             Requires.NotNull(dependenciesByTargetFramework, nameof(dependenciesByTargetFramework));
 
-            if (!activeTargetFramework.Equals(TargetFramework.Empty))
-            {
-                Requires.Argument(
-                    dependenciesByTargetFramework.ContainsKey(activeTargetFramework),
-                    nameof(dependenciesByTargetFramework),
-                    $"Must contain {nameof(activeTargetFramework)} ({activeTargetFramework.FullName}).");
-            }
+            System.Diagnostics.Debug.Assert(
+                activeTargetFramework.Equals(TargetFramework.Empty) || dependenciesByTargetFramework.ContainsKey(activeTargetFramework),
+                $"Active target framework {nameof(activeTargetFramework)} {activeTargetFramework.FullName} {activeTargetFramework.ShortName} " +
```

Comment:
The nameof looks redundant here.. "Active target framework activeTargetFramework Blah"?